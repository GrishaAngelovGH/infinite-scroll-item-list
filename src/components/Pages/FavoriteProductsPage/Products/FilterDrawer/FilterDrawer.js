import { useDispatch, useSelector } from 'react-redux'
import { applyCriteriaFilter } from 'slices/filtersSlice'

import { SearchOutlined } from '@ant-design/icons'
import { Drawer, Input } from 'antd'

const FilterDrawer = ({ open, onClose }) => {
  const criteria = useSelector((state) => state.filters.criteria)
  const dispatch = useDispatch()

  const handleSearch = ({ target }) => {
    dispatch(applyCriteriaFilter(target.value.toLowerCase()))
  }

  return (
    <Drawer
      title='Filter Favorite Product Items'
      placement={'left'}
      closable={true}
      onClose={onClose}
      open={open}
      key={'left'}
    >
      <Input
        value={criteria}
        size='large'
        placeholder='Search'
        prefix={<SearchOutlined />}
        onChange={handleSearch}
      />
    </Drawer>

  )
}

export default FilterDrawer