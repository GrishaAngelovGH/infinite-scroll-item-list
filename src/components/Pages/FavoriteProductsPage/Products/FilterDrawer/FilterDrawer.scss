.price-filter {
  margin: 15px;
}

.title {
  text-decoration: underline;
  padding-top: 10px;
}

.status {
  position: absolute;
  bottom: 5%;
  left: 50%;
  transform: translate(-50%, 0);
}
