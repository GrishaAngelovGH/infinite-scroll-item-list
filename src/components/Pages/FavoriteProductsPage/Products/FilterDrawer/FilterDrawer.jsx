import { useDispatch, useSelector } from 'react-redux'
import { applyCriteriaFilter, applyPriceFilter } from 'slices/filtersSlice'

import {
  selectFavoriteItemRowsByFilter,
  createRows
} from 'selectors/itemRowsSelector'

import { SearchOutlined } from '@ant-design/icons'
import { Drawer, Input, Slider, Tag } from 'antd'

import './FilterDrawer.scss'

const FilterDrawer = ({ open, onClose }) => {
  const criteria = useSelector((state) => state.filters.criteria)
  const price = useSelector((state) => state.filters.price)
  const filteredItemsCount = createRows(useSelector(selectFavoriteItemRowsByFilter)).flat().length
  const dispatch = useDispatch()

  const handleSearch = ({ target }) => {
    dispatch(applyCriteriaFilter(target.value.toLowerCase()))
  }

  const handlePriceChange = price => {
    dispatch(applyPriceFilter(price))
  }

  return (
    <Drawer
      title='Filter Favorite Product Items'
      placement={'left'}
      closable={true}
      onClose={onClose}
      open={open}
      key={'left'}
    >
      <h3 className='title'>Filter By Criteria</h3>
      <Input
        value={criteria}
        size='large'
        placeholder='Search'
        prefix={<SearchOutlined />}
        onChange={handleSearch}
      />

      <h3 className='title'>Filter By Price (${price})</h3>
      <Slider
        className='price-filter'
        marks={{ 100: '100', 1000: '1000' }}
        value={price}
        min={100}
        max={1000}
        step={50}
        onChange={handlePriceChange}
      />

      <div className='status'>
        <Tag color='#108ee9'>
          <h2>Found Items: {filteredItemsCount}</h2>
        </Tag>
      </div>
    </Drawer>
  )
}

export default FilterDrawer