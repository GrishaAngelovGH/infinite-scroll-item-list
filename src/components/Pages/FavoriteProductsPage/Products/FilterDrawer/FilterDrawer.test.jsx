import { render } from '@testing-library/react'

import { Provider } from 'react-redux'

import FilterDrawer from './FilterDrawer'

test('should render FilterDrawer component', () => {
  const view = render(
    <Provider store={globalThis.mockStore}>
      <FilterDrawer open={true} />
    </Provider>
  )

  expect(view).toMatchSnapshot()
})