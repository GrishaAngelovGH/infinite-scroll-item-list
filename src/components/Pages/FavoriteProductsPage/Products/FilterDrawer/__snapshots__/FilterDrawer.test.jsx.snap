// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render FilterDrawer component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div>
      <div
        class="ant-drawer ant-drawer-left css-dev-only-do-not-override-19lec04 ant-drawer-open"
        tabindex="-1"
      >
        <div
          class="ant-drawer-mask"
        />
        <div
          aria-hidden="true"
          data-sentinel="start"
          style="width: 0px; height: 0px; overflow: hidden; outline: none; position: absolute;"
          tabindex="0"
        />
        <div
          class="ant-drawer-content-wrapper"
          style="width: 378px;"
        >
          <div
            aria-modal="true"
            class="ant-drawer-content"
            role="dialog"
          >
            <div
              class="ant-drawer-header"
            >
              <div
                class="ant-drawer-header-title"
              >
                <button
                  aria-label="Close"
                  class="ant-drawer-close"
                  type="button"
                >
                  <span
                    aria-label="close"
                    class="anticon anticon-close"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close"
                      fill="currentColor"
                      fill-rule="evenodd"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
                      />
                    </svg>
                  </span>
                </button>
                <div
                  class="ant-drawer-title"
                >
                  Filter Favorite Product Items
                </div>
              </div>
            </div>
            <div
              class="ant-drawer-body"
            >
              <h3
                class="title"
              >
                Filter By Criteria
              </h3>
              <span
                class="ant-input-affix-wrapper ant-input-affix-wrapper-lg css-dev-only-do-not-override-19lec04 ant-input-outlined"
              >
                <span
                  class="ant-input-prefix"
                >
                  <span
                    aria-label="search"
                    class="anticon anticon-search"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="search"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                      />
                    </svg>
                  </span>
                </span>
                <input
                  class="ant-input ant-input-lg css-dev-only-do-not-override-19lec04"
                  placeholder="Search"
                  type="text"
                  value=""
                />
              </span>
              <h3
                class="title"
              >
                Filter By Price ($
                1000
                )
              </h3>
              <div
                class="ant-slider price-filter css-dev-only-do-not-override-19lec04 ant-slider-horizontal ant-slider-with-marks"
              >
                <div
                  class="ant-slider-rail"
                />
                <div
                  class="ant-slider-track"
                  style="left: 0%; width: 100%;"
                />
                <div
                  class="ant-slider-step"
                >
                  <span
                    class="ant-slider-dot ant-slider-dot-active"
                    style="left: 0%; transform: translateX(-50%);"
                  />
                  <span
                    class="ant-slider-dot ant-slider-dot-active"
                    style="left: 100%; transform: translateX(-50%);"
                  />
                </div>
                <div
                  aria-describedby="test-id"
                  aria-disabled="false"
                  aria-orientation="horizontal"
                  aria-valuemax="1000"
                  aria-valuemin="100"
                  aria-valuenow="1000"
                  class="ant-slider-handle"
                  role="slider"
                  style="left: 100%; transform: translateX(-50%);"
                  tabindex="0"
                />
                <div
                  class="ant-slider-mark"
                >
                  <span
                    class="ant-slider-mark-text ant-slider-mark-text-active"
                    style="left: 0%; transform: translateX(-50%);"
                  >
                    100
                  </span>
                  <span
                    class="ant-slider-mark-text ant-slider-mark-text-active"
                    style="left: 100%; transform: translateX(-50%);"
                  >
                    1000
                  </span>
                </div>
              </div>
              <div
                class="status"
              >
                <span
                  class="ant-tag ant-tag-has-color css-dev-only-do-not-override-19lec04"
                  style="background-color: rgb(16, 142, 233);"
                >
                  <h2>
                    Found Items: 
                    2
                  </h2>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          data-sentinel="end"
          style="width: 0px; height: 0px; overflow: hidden; outline: none; position: absolute;"
          tabindex="0"
        />
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
